JSON Parser

gboldi19 opened the pull request:
@wajzy elkészítettem a JSON parsert, ami három féle bemenetet vár (fajlnev, istream, string), map-eken keresztül. Nem feltételeztem azt, hogy a parseolni kívánt fájlunk helyes, hanem vizsgáltam a kapcsos zárójelek, idézőjelek, sorvégi karakterek meglétét vagy hiányát, emiatt van ennyi exception. UNIT testek még készülőben vannak, legkésőbb ma estére tervezem befejezni. Ezt leszámítva a kód többi részében esetleg van valami hiba vagy egyéb észrevétel?

gboldi19 committed:
Preparating Character header and .cpp to accept JSONParser

gboldi19 committed:
JSONParser class added

Added 3 type parsing (input, stream, file) with proper exception handling

gboldi19 committed:
Preparing .gitignore to handle folders

gboldi19 committed:
Added workflow handler inclusion

gboldi19 committed:
.gitignore updated for unit parser testing

gboldi19 committed:
Bad templates for unit testing

gboldi19 committed:
Fixed workflow naming

gboldi19 committed:
Matching .yml name to its function

gboldi19 committed:
Bypassing a cache related problem

gboldi19 committed:
Renamed first task workflow to be more precise

gboldi19 committed:
Bypass solution reverted

gboldi19 committed:
gitignore updated

gboldi19 committed:
Preparing a CMakeList

gboldi19 committed:
Google test parse handled

gboldi19 committed:
fixing a type-o

gboldi19 committed:
Include fix to .cpp

gboldi19 committed:
Fixing minor issues

gboldi19 committed:
Beautyfing .cpp

gboldi19 committed:
Proper map insertion

gboldi19 committed:
Fixing cmakelists

gboldi19 committed:
Fixed texting and method order, aligning

gboldi19 committed:
Removed private declaration

gboldi19 committed:
Interesting anomaly regarding to CMakeLists.txt

gboldi19 committed:
Shortened unit testing

gboldi19 committed:
Fixed a type-o

gboldi19 commented:
@wajzy unit test készen van és le kéne futnia, azonban a pcloudra felkerült videóban is említésre került, hogy a google test és makrói leginkább c++11-re lettek tervezve. A workflow azzal száll el, hogy a `map<std::string, std::any>`-ben az "any" map template-t nem találja az std-ben: `error: ‘any’ is not a member of ‘std’
 std::map<std::string, std::any>`. 
Az std::any map template a hivatalos dokumentáció szerint c++17-től van értelmezve, és bár annak ellenére, hogy erre buildeltünk a google test képtelen értelmezni

gboldi19 committed:
Removing not used test files

gboldi19 committed:
Adding compile features

gboldi19 committed:
Renaming executable

gboldi19 committed:
Including .cpp instead of header

gboldi19 committed:
Debugging changes, class to namespace

gboldi19 committed:
Removed public as its not a class

gboldi19 committed:
Removed redefinition

gboldi19 committed:
.gitignore fixed

gboldi19 committed:
Removed accidentally added file

gboldi19 committed:
Merge branch 'jsonparser' of https://github.com/Teaching-projects/SZE-MOSZE-2020-Mortar_Kombat into jsonparser

gboldi19 committed:
removed .cpp including

gboldi19 committed:
Header protection to Character

gboldi19 committed:
Fixed including Character header

gboldi19 committed:
Removed ../ from json_parser_test

gboldi19 committed:
Header linking debug

gboldi19 committed:
Header linking fix #2

gboldi19 committed:
Header debug

gboldi19 committed:
Header fixing

gboldi19 committed:
Removing unused inclusions

gboldi19 committed:
Removing static

gboldi19 committed:
Referencing root directory

gboldi19 committed:
Removed memory address reference

gboldi19 committed:
Reverting change

gboldi19 commented:
Build és compile is lefut, több alkalmazott változtatás sem hozott eredményt eddig sajnos, még mindig azt írja, hogy `‘any’ is not a member of ‘std’ std::map<std::string, std::any>`, amikor unit tesztelni szeretném, valószínüleg valamelyik headert nem látja.

wajzy commented:
Pedig ezt bizony le lehet fordítani, és lehet tesztelni.
[any.zip](https://github.com/Teaching-projects/SZE-MOSZE-2020-Mortar_Kombat/files/5397569/any.zip)
A szokásos parancsokkal feltámasztható:
```
cmake CMakeLists.txt
make
./runTests
```

wajzy commented:
@gboldi19 Jelezz, ha már futnak a tesztek rendesen, utána megnézem.

gboldi19 committed:
Fixed linker problem, now calls std::any correctly

gboldi19 committed:
Fixed a bracket missing

gboldi19 committed:
Proper Google Testing, added ASSERT_THROW

gboldi19 commented:
@wajzy végre sikerült lefuttatni a teszteket, és elfogadta az any_cast-ot is a mapon belül. Köszönöm szépen a tippet!

wajzy reviewed (approved):
Ez annyira összetett lett, egyszerűen jónak kell lennie :)

wajzy commented:
Kérek egy kis conflict feloldást!

gboldi19 committed:
Conflicts solved, workflow push style changed

gboldi19 committed:
Fixed updated output

gboldi19 committed:
It can help sometimes if u dont comment out the code you'd like to use

gboldi19 committed:
Type-o fix

gboldi19 committed:
Merge branch 'master' into jsonparser

gboldi19 committed:
Update and rename tests.yml to memtest_analysis.yml

gboldi19 committed:
Fixed workflow naming

gboldi19 committed:
Memtest analysis inclusion error fix

gboldi19 committed:
Memtest analysis inclusion error fix #2

balazs482 committed:
Warning log file artifacts.

Warning and error messages now artifacts too for testing purposes. To be rolled back.

balazs482 committed:
"exit 1" line commented out.

Temporarily removed exit if errors and/or warnings were found. Now the workflow can artifact. To be rolled back.

gboldi19 committed:
Fixed "return value of valueString.substr() not used" error

gboldi19 committed:
Reverting a change that were being made for debugging

gboldi19 committed:
Reverting a change related to debugging

gboldi19 committed:
Reverting a change related to debugging

gboldi19 commented:
@wajzy sikerült a conflictokat feloldani és egy bent felejtett bug-ot kijavítani (Commit ID: 5d3bb41f7b44073488a7bef37d8dc8d154ab632f)
